### PR TITLE
split Directory.Build file into two dedicated files, one for src (packages) and one for test projects

### PR DESCRIPTION
**Pull request by adamsitnik**
#2185 inspired me to do a little bit of MSBuild refactor: split Directory.Build file into two dedicated files, one for src (packages) and one for test projects. This allows us the reduce the number of conditions in the props file

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
+    <PackageProjectUrl>https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks</PackageProjectUrl>
+    <RepositoryUrl>https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks</RepositoryUrl>
+    <Authors>Xabaril Contributors</Authors>
+    <Company>Xabaril</Company>
+    <RepositoryType>git</RepositoryType>
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
+    <PackageTags>HealthCheck;HealthChecks;Health</PackageTags>
+    <NoWarn>$(NoWarn);1591;IDISP013;AD0001;</NoWarn> <!--TODO: temporary solution, remove AD0001 after https://github.com/dotnet/aspnetcore/issues/50836 fixed-->
+    <CheckEolTargetFramework>false</CheckEolTargetFramework>
+    <SuppressTfmSupportBuildWarnings>true</SuppressTfmSupportBuildWarnings>
```
Comment (eerhardt) [inline src/Directory.Build.props:19]: This is still in the root Directory.Build.props file. Can be removed here.

File: src/Directory.Build.props [hunk 2]
```diff
@@ -1,10 +1,39 @@
...
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
+    <PackageProjectUrl>https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks</PackageProjectUrl>
+    <RepositoryUrl>https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks</RepositoryUrl>
+    <Authors>Xabaril Contributors</Authors>
+    <Company>Xabaril</Company>
+    <RepositoryType>git</RepositoryType>
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
+    <PackageTags>HealthCheck;HealthChecks;Health</PackageTags>
+    <NoWarn>$(NoWarn);1591;IDISP013;AD0001;</NoWarn> <!--TODO: temporary solution, remove AD0001 after https://github.com/dotnet/aspnetcore/issues/50836 fixed-->
```
Comment (eerhardt) [inline src/Directory.Build.props:17]: This is still in the root Directory.Build.props file. Can be removed here.

**Review by eerhardt — APPROVED** [review]
I had to force myself not to do this as part of my PR. 😆

File: src/Directory.Build.props (hunk 2 above)
Comment (adamsitnik) [inline src/Directory.Build.props:17]:
```suggestion
```

File: src/Directory.Build.props (hunk 1 above)
Comment (adamsitnik) [inline src/Directory.Build.props:19]:
```suggestion
```

**Pull request merged**